rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeperServer.java

```diff
@@ -113,8 +116,14 @@ public static void startServer(Configuration conf, MetricRegistry metricsRegistr
   /**
    * Register desired metrics.
    */
-  private static void registerMetrics()
+  private static void registerMetrics(Configuration conf)
   {
+    if (CacheConfig.isOnMaster(conf)) {
```

Comment:
We need one more config to also control whether statsd runs on the master as well. So the if condition needs to be:

    if (CacheConfig.isOnMaster(conf) && CacheConfig.startStatsDReporter(conf)) {

Actually I suggest the following to be future proof:

    if ((CacheConfig.isOnMaster(conf) && CacheConfig.statsdReportOnMaster(conf)) 
       || (!CacheConfig.isOnMaster(conf) && CacheConfig.statsdReportOnWorker(conf))) {

Reply:
What should the default values for these flags be? My thoughts:
```
.isOnMaster -> false (1 master vs. multiple workers)
.reportOnMaster -> true
.reportOnWorker -> false
```

Reply:
`reportOnMaster` should also be `false`. We will roll it out gradually through QDS cluster configuration.

Reply:
- Flags added to CacheConfig
- Unit tests added for verifying metrics are reported/not reported based on flag combinations
- Flag combinations tested on Qubole cluster